#5445 Change image save path for ExplorationFileSystem to be consiste…

sstone-codaio opened the pull request.

seanlip commented on core/domain/exp_jobs_one_off.py:
```diff
@@ -59,7 +59,7 @@
     feconf.ACCEPTED_IMAGE_FORMATS_AND_EXTENSIONS.values()))
 FILE_MODEL_ID_REGEX = re.compile(
-    r'^/([^/]+)/assets/(([^/]+)\.(' + '|'.join(
+    r'^/([^/]+)/assets/image/(([^/]+)\.(' + '|'.join(
```
Just wondering, should this match GCS_IMAGE_ID_REGEX fully?

More generally, the more parity between GCS and local datastore we could have, the better. Ideally anything outside fs_domain has no idea what filesystem it's writing to, it should all just behave the same way.

sstone-codaio replied:
Good point! I think we can remove FILE_MODEL_ID_REGEX.

seanlip commented on core/domain/exp_jobs_one_off.py:
```diff
@@ -879,7 +879,7 @@ def map(file_model):
             else:
                 filename = catched_groups.group(2)
-                filepath = 'assets/' + filename
+                filepath = 'assets/image/' + filename
```
As an FYI I think we can just go ahead and delete this entire job. It's already been run in production and is no longer needed because there are no more files in the production datastore.

sstone-codaio replied:
Thanks Sean, just to confirm, do you mean we don't need to run any of the one off jobs defined here in the future?

sshou14 replied:
Thanks Sean, just to confirm, do you mean we don't need any of the one off jobs defined here in the future?

seanlip replied:
We don't need to run DeleteImagesFromGAEJob, ValidationOfImagesOnGCSJobUsingExps or ImageDataMigrationJob in the future; the other jobs may still be needed. But these three jobs (and the tests for them) can be deleted, I think. Not sure if that means that the constant at the top is now unused, but if so we can delete it too.

Thanks!

sstone-codaio replied:
Thanks for the clarification, I've removed these three classes in the latest change.

seanlip commented on core/domain/exp_jobs_one_off.py:
```diff
@@ -665,57 +662,6 @@ def reduce(key, values):
-            yield (status, '%s' % set(values))
-
-
 class ValidationOfImagesOnGCSJob(jobs.BaseMapReduceOneOffJobManager):
```
Actually let's get rid of this job too. I realize it is based on FileModel which no longer exists (and replacing FILE_MODEL_ID_REGEX with GCS_IMAGE_ID_REGEX is thus conceptually incorrect). Sorry for not catching this earlier.

sstone-codaio replied:
Sounds good Sean

seanlip commented on core/domain/fs_domain.py:
```diff
@@ -165,10 +165,10 @@ def _get_file_metadata(self, filepath, version):
         if version is None:
             return file_models.FileMetadataModel.get_model(
-                self._exploration_id, 'assets/%s' % filepath)
+                self._exploration_id, 'assets/image/%s' % filepath)
```
I just realized that this might not be the correct fix. Could you check what happens when you upload an audio file for an exploration locally? Where does it get stored? (Its path should be assets/audio/... but that seems impossible with this change.)

Since the aim is to have a dev file system that has parity with GCS (see GcsFileSystem below) the APIs ought to work the same way.

sstone-codaio replied:
Thanks Sean, you're right. 
I've kept the API unchanged and added a method in fs_services in the latest change.

Commit:
Remove three unused classes from exp_jobs_one_off.py

Removed DeleteImagesFromGAEJob, ValidationOfImagesOnGCSJobUsingExps, and
ImageDataMigrationJob according to code review.

Commit:
Fix #5445:Standardise the two filesystems (ExplorationFileSystem and GCSFileSystem)

Commit:
Address code review comment to remove the ValidationOfImagesOnGCSJob class

Commit:
fix merge conflict with remove develop

Commit:
Address code review comments

seanlip commented:
Hi @sshou14, just a note: please could you reply to all comments, following the instructions in step 5 [here](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change)? In particular, for comments that you've addressed, please reply "Done." -- this helps keep track of which comments have been addressed and which haven't. Thanks!

(This would also signify that the PR is ready for another review.)

Commit:
Address comments to consolidate endpoint url for image and audio assets

sstone-codaio reviewed (commented):
I removed FILE_MODEL_ID_REGEX as no code is referencing it

sstone-codaio commented:
@seanlip Sorry for the late reply Sean, this pr is ready for review.

sstone-codaio committed:
Merge branch 'develop' into sshou-fix5445

seanlip commented on core/templates/dev/head/services/AssetsBackendApiService.js:
```diff
@@ -31,16 +35,14 @@ oppia.factory('AssetsBackendApiService', [
     var ASSET_TYPE_AUDIO = 'audio';
     var ASSET_TYPE_IMAGE = 'image';
 
+    GCS_PREFIX = ('https://storage.googleapis.com/' +
```
missing `var`?

sstone-codaio replied:
Thanks, good catch

seanlip commented on core/templates/dev/head/services/AssetsBackendApiService.js:
```diff
@@ -31,16 +35,14 @@ oppia.factory('AssetsBackendApiService', [
-        '/audiohandler/<exploration_id>/audio/<filename>');
+      (DEV_MODE ? '/audiohandler' : GCS_PREFIX) +
+      '/<exploration_id>/assets/audio/<filename>');
     var IMAGE_DOWNLOAD_URL_TEMPLATE = (
```
Long run, might be good to consolidate the "dev" part into /assetsdevhandler/<exploration_id>/assets/.../<filename>. Would you agree?

If so, maybe do in a separate PR?

sstone-codaio replied:
I agree, so I'll create an issue and assign to myself?

sstone-codaio replied:
Created #5713

seanlip reviewed (approved):
Thanks @sshou14! It looks good to me. Left a couple comments -- please address the first and respond to the other (maybe file and self-assign an issue if you agree?), then it's fine to get this merged.

Commit:
Address code review

Commit:
Merge branch 'sshou-fix5445' of https://github.com/sshou14/oppia into sshou-fix5445

seanlip commented:
LGTM. Thanks @sshou14!
